evmcore/validation_test.go

```diff
@@ -0,0 +1,774 @@
+			require.ErrorIs(t, err, ErrNonceTooLow)
+		})
+	}
+}
+
+func TestValidateTx_Value_RejectsTxWith(t *testing.T) {
+	for name, tx := range getTxsFromAllTypes() {
+		t.Run(fmt.Sprintf("negative value/%v", name), func(t *testing.T) {
+			if isBlobOrSetCode(tx) {
+				t.Skip("blob and setCode transactions cannot have negative value because they use uint256 Value")
+			}
+			setValueToNegative(t, tx)
+			err := validateTx(types.NewTx(tx), types.NewPragueSigner(big.NewInt(1)),
+				testTransactionsOption())
+			require.ErrorIs(t, err, ErrNegativeValue)
+		})
+	}
+}
+
+func TestValidateTx_GasPriceAndTip_RejectsTxWith(t *testing.T) {
+	extremelyLargeN := new(big.Int).Lsh(big.NewInt(1), 256)
+
+	// GasPrice/GasFeeCap tests
+	for name, tx := range getTxsFromAllTypes() {
+		t.Run(fmt.Sprintf("gas fee longer than 256 bits/%s", name), func(t *testing.T) {
+			if isBlobOrSetCode(tx) {
+				t.Skip("blob and setCode transactions cannot have gas price larger than uint256")
+			}
+			setGasPriceOrFeeCap(t, tx, extremelyLargeN)
+			err := validateTx(types.NewTx(tx), types.NewPragueSigner(big.NewInt(1)),
+				testTransactionsOption())
+			require.ErrorIs(t, err, ErrFeeCapVeryHigh)
+		})
+	}
+
+	for name, tx := range getTxsFromAllTypes() {
+		t.Run(fmt.Sprintf("gas price lower than base fee/%v", name), func(t *testing.T) {
+			// setup validation context
+			opt := testTransactionsOption()
+			opt.currentBaseFee = big.NewInt(2)
+
+			// gas fee cap should be higher than current gas price
+			setGasPriceOrFeeCap(t, tx, big.NewInt(1))
+
+			// sign txs with sender
+			signer, _, signedTx := signTxForTest(t, tx)
+
+			// validate transaction
+			err := validateTx(signedTx, signer, opt)
+			require.ErrorIs(t, err, ErrUnderpriced)
+		})
+	}
+
+	// GasTipCap test
+	for name, tx := range getTxsFromAllTypes() {
+		t.Run(fmt.Sprintf("gas tip longer than 256 bits/%s", name), func(t *testing.T) {
+			// Legacy and access list transactions do not conceptually have a tip.
+			if isLegacyOrAccessList(tx) {
+				t.Skip("legacy and access list transactions do not have gas tip")
+			}
+			// Blob and setCode transactions can never have a Tip with bit length
+			// larger than 256 because of the type they use for this field.
+			if isBlobOrSetCode(tx) {
+				t.Skip("blob and setCode transactions cannot have gas tip larger than u256")
+			}
+
+			// set gas tip cap too large
+			setGasTipCap(t, tx, extremelyLargeN)
+
+			err := validateTx(types.NewTx(tx), types.NewPragueSigner(big.NewInt(1)),
+				testTransactionsOption())
+
+			if _, ok := tx.(*types.DynamicFeeTx); ok {
+				require.ErrorIs(t, err, ErrTipVeryHigh)
+			} else {
+				t.Fatal("unknown transaction type")
+			}
+		})
+	}
+
+	for name, tx := range getTxsFromAllTypes() {
+		t.Run(fmt.Sprintf("gas tip lower than pool min tip/%v", name), func(t *testing.T) {
+			// legacy and access list transactions do not have tip cap
+			if isLegacyOrAccessList(tx) {
+				t.Skip("legacy and access list transactions do not have tip cap")
```

Comment:
Please check whether the gas-price field serves the role of the gas-tip-cap for these types of transactions.

Reply:
skip removed in https://github.com/0xsoniclabs/sonic/pull/154/commits/2bcc9f0c1d81257091eec396d0f790809ad9eae1